app/assets/javascripts/application.js.erb

```diff
@@ -515,16 +465,41 @@ const onPageLoad = (callback) => {
 const replace = (oldElement, newHTMLString) => {
   const placeHolder = document.createElement('div');
   placeHolder.insertAdjacentHTML('afterBegin', newHTMLString);
-  const newElement = placeHolder.firstElementChild;
-  oldElement.replaceWith(newElement);
+  const newElements = placeHolder.childNodes.values();
+  oldElement.replaceWith(...newElements);
+}
+
+const fetchWithCSRF = (url, params) => {
+  let csrfToken = document.querySelector('meta[name="csrf-token"]').getAttribute('content');
+  let h = new Headers();
+  h.append('X-CSRF-Token', csrfToken);
+  params = params || {};
+  params['headers'] = h;
+  return fetch(url, params);
+} 
+
+const removeExtraInput = (targetElement) => {
```

Comment:
I see how this function works to fix the template issue, but it's really brittle in this form. If the template is fixed to not render an extra `<input>`, it'll remove the working input. If multiple rerenders of the comment happen before it's called, it'll only remove one.

Let's be more specific in the selector: the goal is to find _extra_ inputs, not all:

```
const extraInputs = targetElement.parentElement.querySelectorAll('.comment_folder_button + .comment_folder_button');
for (const i of extraInputs) {
  i.remove();
}
```

Reply:
This is a much better function, as it removes the dependency on the error to not BE an error.